IdentityCore/src/oauth2/aad_base/MSIDAADTokenResponse.m

```diff
@@ -161,18 +107,67 @@ - (NSDictionary *)additionalServerInfo
                              MSID_OAUTH2_EXT_EXPIRES_IN, @"url",
                              MSID_OAUTH2_SUB_ERROR];
     
-    additionalInfo = [additionalInfo dictionaryByRemovingFields:knownFields];
-    if (additionalInfo.count > 0)
+    NSDictionary *additionalInfo = [additionalServerInfo dictionaryByRemovingFields:knownFields];
+    
+    [super setAdditionalServerInfo:additionalInfo];
+}
+
+#pragma mark - MSIDJsonSerializable
+
+- (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)error
+{
+    self = [super initWithJSONDictionary:json error:error];
+    if (self)
     {
-        return additionalInfo;
+        _correlationId = [json msidStringObjectForKey:MSID_OAUTH2_CORRELATION_ID_RESPONSE];
+        _familyId = [json msidStringObjectForKey:MSID_FAMILY_ID];
+        _speInfo = [json msidStringObjectForKey:MSID_TELEMETRY_KEY_SPE_INFO];
+        _suberror = [json msidStringObjectForKey:MSID_OAUTH2_SUB_ERROR];
+        _additionalUserId = [json msidStringObjectForKey:@"adi"];
+        
+        NSString *rawClientInfo = [json msidStringObjectForKey:MSID_OAUTH2_CLIENT_INFO];
+        NSError *localError;
+        _clientInfo = [[MSIDClientInfo alloc] initWithRawClientInfo:rawClientInfo error:&localError];
+        if (localError) MSID_LOG_WITH_CTX_PII(MSIDLogLevelError, nil, @"Failed to init client info, error: %@", MSID_PII_LOG_MASKABLE(localError));
```

Comment:
I think clientInfo is non-critical. There're edge cases in AAD when we don't receive it, and our auth flows shouldn't be failing in those.

Reply:
done